### PR TITLE
issue/2 Check all previous locks

### LINKED ISSUE #2: Locking should not lock completed items
For example: sequential locking requires the completion of the previous item. If the current item is already complete, it should not be locked.

This is particularly awkward if a user rests completion on a component in a previous topic. It then causes subsequent menu items to relock even when previously completed.

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #2 

### Fixed
* Locking is aware of previous locking state, if the previous item _isLocked then this item will be locked

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by eleanor-heath — APPROVED** [review]

**Review by danielghost — COMMENTED** [review]
This code works as explained but I think this may cause other UX issues as explained here:
https://github.com/adaptlearning/adapt-contrib-core/issues/2#issuecomment-916019260

**Comment by oliverfoster** [comment]
I think we have agreed that all previous locks should be checked rather than just the last previous lock. PR needs modifying.

**Comment by oliverfoster** [comment]
Updated according to https://github.com/adaptlearning/adapt-contrib-core/issues/2#issuecomment-916121035

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by eleanor-heath — APPROVED** [review]

File: js/models/adaptModel.js [hunk 1]
```diff
@@ -749,8 +755,13 @@ export default class AdaptModel extends LockingModel {
     if (!lockedBy) return false;
     return lockedBy.some(id => {
       try {
-        const model = Adapt.findById(id);
-        return !model.get('_isComplete') && !model.get('_isOptional') && model.get('_isAvailable');
+        const anotherModel = Adapt.findById(id);
+        return anotherModel.get('_isLocked') ||
```
Comment (oliverfoster) [inline js/models/adaptModel.js:759]: Is it worth putting an availability check in here?

Comment (oliverfoster) [reply js/models/adaptModel.js]: Done

**Review by joe-allen-89 — APPROVED** [review]
👀

**Pull request merged**